### PR TITLE
Remove unused legacy splash screen content.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Remove unused legacy splash screen content.
Addresses issue #2290

## Relevant technical choices

N/A

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/dashboard-splash/DashboardSplashApp.js [hunk 1]
```diff
@@ -23,16 +23,16 @@ import '../publisher-wins';
...
 const { useSelect } = Data;
 
 export default function DashboardSplashApp() {
 	const usingProxy = useSelect( ( select ) => select( CORE_SITE ).isUsingProxy() );
 
 	if ( usingProxy === true ) {
 		return <SetupUsingProxy />;
-	} else if ( usingProxy === false ) {
-		return <LegacyDashboardSplashApp />;
+	} if ( usingProxy === false ) {
```
Comment (tofumatt) [inline assets/js/components/dashboard-splash/DashboardSplashApp.js:34]: Suuuper minor, but let's use a newline here.

```suggestion
	}
	
	if ( usingProxy === false ) {
```

**Review by tofumatt — APPROVED** [review]
Oddly, the IB mentions needing to approve storybook/VRT changes, but I didn't see them here. Suspecting that step was missing, I checked out this branch locally and ran the tests without failure.

Since the stories were removed, it seems like they weren't being used for VRT so it's fine to ignore that part of the IB.

Otherwise: looks good to me! 👍🏻

**Review by tofumatt — DISMISSED** [review]

**Review by aaemnnosttv — APPROVED** [review]
👍 👍

**Review by felixarntz — APPROVED** [review]
Great!

So wonderful 😍 
Screenshot 2020-12-14 at 14 14 35

**Pull request merged**